lib/history/memory/inmemory.go

```diff
@@ -102,3 +112,16 @@ func (t *Timeline) getFilteredEvents(params map[string]string) (events []*pb.Tim
 	// TODO: for now just return the unfiltered events.
 	return t.events
 }
+
+// filterDuplicates removes duplicate events.
+func (t *Timeline) filterDuplicates() {
+	set := make(map[*pb.TimelineEvent]struct{})
```

Comment:
Hm, so what is a "duplicate" event in this case? Since this holds a pointer to a TimelineEvent, it means that a duplicate is exact same event pointer? When/how can this happen?

Reply:
Yeah, so this was a bug.
In order to compare TimelineEvents they first need to be converted into a comparable struct.